math/n_bonacci.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file
+ * @brief Implementation of [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers)
+ * Series
+ *
+ * @details
+ * In general, in N-bonacci sequence,
+ * we generate sum of preceding N numbers from the next term.
+ *
+ * For example, a 3-bonacci sequence is the following:
+ * 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ * In this code we take N and M as input where M is the number of terms
+ * to be printed of the N-bonacci series
+ *
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+#include <algorithm>  /// for std::is_sorted, std::swap
+#include <cassert>    /// for assert
+#include <iostream>   /// for io operations
+#include <vector>     /// for std::vector
+
+namespace math {
+/**
+ * @namespace math
+ * @brief Function for n-bonacci series
+ */
+namespace n_bonacci {
+/**
+ * @brief The main function implements n-bonacci series
+ * @tparam T type of array
+ * @param n,m m is the number of terms in n-bonacci sequence
+ * @returns the n-bonacci sequence as vector array
+ */
+template <typename T>
+std::vector<T> N_bonacci(const std::vector<T> &in_arr) {
```

Comment:
Would you please justify the use of `in_arr` in this case? It seems like you can simply use two `int` vars `n` and `m` and that will serve the purpose well. Also vars `n` and `m` will have non negative integral value in all the cases therefore instead of using template argument it's better to use signed or unsinged `int`.

Reply:
I actually did it using n and m as parameters initially but while building the file it showed error in test() func as "n is not declared" , "error: expected unqualified-id before ‘unsigned’"... so I thought using an array of size two would be easier that is why in commented part(which i wrote initially) has parameters as n and m.

Reply:
Yeah, actually I started with n and m but because of many errors coming I thought array of size 2 will work easier. But now I have changed it and have removed in_arr and added n,m as params @ayaankhan98 see it once.